### DISCUSSION TITLE
Coloring Solver Crashes

### DISCUSSION THREAD

**Discussion by Otherside5** [mdolab/dafoam#136] [General]
Hello everyone,

I am trying to optimise a centrifugal compressor with DATurboFoam based on the rotor 37 tutorial. Unfortunately when I want to run the optimisation it crashes after "Checking if Coloring file exists.." with a segmentation fault.

This is the logOpt file it generates maybe this helps: [logOpt.txt](https://github.com/mdolab/dafoam/files/6448603/logOpt.txt)

The unstructured mesh I am using was generated with gmsh and tested with Openfoam v1906. It also works in DAFoam when I set the task to "runPrimal". I tried to keep as much of the settings as possible from the tutorial but I could not isolate the error so far.

I would be very grateful for any suggestions

Regards Alex

**Comment by friedenhe** [comment]
The problem is likely related to the AMI boundary condition (we don't support that in the current version). That is why in the rotor37 tutorial, we use the cyclic BC.

**Reply by Otherside5** [reply]
Thank you for your quick reply!

I tried to replace the cyclicAMI BC with wall BC and also improved my mesh to use cyclic as in the tutorial but the result was the same as before in both cases.

In the end the problem was that I was running the case on a single core because with AMI the decomposing did not work. I figured this out when I went back to the tutorial case and it did not run anymore. With cyclic BC and on 4 cores everything is running fine so far.

Thanks again
Alex

**Reply by hosseinnegahban** [reply]
Hi Alex,

I have encountered the same problem as yours, the solution stops at "Checking if coloring file exists..". I have tried parallel with 4 core  and also single core but no solution so far. Do you have any recommendation?

Best regards,
Hossein

**Reply by Otherside5** [reply]
Hello Hossein,

in my case all problems came with the periodic BC. The only hints I can give you is to make sure that your mesh/geometry is perfectly periodic (this was my biggest problem) and maybe use as much cores as you have.

Best wishes
Alex

**Reply by alinaeem121212** [reply]
Could you please guide me on how you resolved this error?
I am not using periodic boundary conditions, it is an external flow over a body.

**Comment by friedenhe** [comment]
This is an older post, but @alinaeem121212, you may want to check whether the issue is related to insufficient system memory.
